avaframe/ana1Tests/energyLineTest.py

```diff
@@ -0,0 +1,332 @@
+               color='r', linestyle='--')
+    ax3.hlines(y=zIntersection-z0, color='b', xmin=sMin, xmax=sIntersection, linestyle='--')
+
+    ax3.set_xlabel('s [m]')
+    ax3.set_ylabel(r'$\Delta z$ [m]')
+    ax3.yaxis.set_label_coords(0, 0.9)
+    ax3.set_xlim([sMin, sMax])
+    ax3.set_ylim([zMin, zMax])
+    ax3.tick_params(axis='x', which='major', labelsize=8, rotation=45)
+    ax3.tick_params(axis='y', which='major', labelsize=8, rotation=45)
+    ax3.set_xticks([avaProfileMass['s'][-1], sIntersection])
+    ax3.set_yticks([avaProfileMass['z'][-1]-z0, zIntersection-z0])
+    ax3.xaxis.set_major_formatter(FormatStrFormatter('%.2f'))
+    ax3.yaxis.set_major_formatter(FormatStrFormatter('%.2f'))
+    text = ('Run-out s diff : %.2f m \nRun-out z diff : %.2f m \nRun-out angle diff : %.4f° \nvelocity height rmse : %.2f m \n(energy line - alpha line)' %
+            (runOutSError, runOutZError, runOutAngleError, rmseVelocityElevation))
+    text_box = AnchoredText(text, frameon=False, loc=1, pad=0.5)
+    plt.setp(text_box.patch, facecolor='white', alpha=0.5)
+    ax3.add_artist(text_box)
+    outFileName = '_'.join([simName, 'EnergyTest'])
+    outDir = pathlib.Path(avalancheDir, 'Outputs', 'ana1Tests')
+    pU.saveAndOrPlot({'pathResult': outDir}, outFileName, fig)
+
+
+def getRunOutAngle(avaProfileMass):
+    """Compute the center of mass runout angle
+
+    Parameters
+    -----------
+    avaProfileMass: dict
+        particle mass average properties
+
+    Returns
+    --------
+    runOutAngleRad: float
+        Center of Mass runout angle in radians
+    runOutAngleDeg: float
+        Center of Mass runout angle in degrees
+    """
+    # compute horizontal and vertical traveled distance
+    deltaS = avaProfileMass['s'][-1] - avaProfileMass['s'][0]
+    deltaZ = avaProfileMass['z'][-1] - avaProfileMass['z'][0]
+    # extract simulation run-out
+    runOutAngleRad = np.arctan(np.abs(deltaZ/deltaS))
+    runOutAngleDeg = np.rad2deg(runOutAngleRad)
+    return runOutAngleRad, runOutAngleDeg
+
+
+def getAlphaProfileIntersection(avaProfileMass, mu):
+    """Extend the  profile path and compute the intersection
+    between the theoretical energy line and the path profile
+
+    Parameters
+    -----------
+    avaProfileMass: dict
+        particle mass average properties
+    mu: float
+        friction coefficient
+
+    Returns
+    --------
+    slopeExt: float
+        slope of the extrapolation line
+    sIntersection: float
+        s coord of the intersection between the line of slope mu and the
+        mass average path profile
+    zIntersection: float
+        z coord of the intersection between the line of slope mu and the
+        mass average path profile
+    coefExt: float
+        coefficient saying how long the path was extended to find the intersection
+    """
+    # get slope of the last profile points to extend the profile
+    p1 = np.polyfit(avaProfileMass['s'][-3:], avaProfileMass['z'][-3:], 1)
```

Comment:
should the -3 go to an ini file? how many points should this maybe depend on the meshsize?

Reply:
I changed this, Only looking at the two last points. If we decide to change it Ill make an ini file

Reply:
hmm I think isn't this a parameter and would be good to have it in the ini file as it is a parameter?